checker/src/utils.rs

```diff
@@ -58,6 +59,139 @@ pub fn is_public(def_id: DefId, tcx: &TyCtxt<'_, '_, '_>) -> bool {
     }
 }
 
+/// Returns a string that is a valid identifier, made up from the concatenation of
+/// the string representationss of the given list of argument types.
+pub fn argument_types_key_str<'tcx>(tcx: &TyCtxt<'_, '_, 'tcx>, ty: Ty<'tcx>) -> String {
```

Comment:
There may be ambiguities here, right? Is that a potential problem?

Reply:
It is a potential problem. Let me make it less likely by ensuring that it always violates a style rule.